src/test/json/json_value_test.cpp

```diff
@@ -17,6 +17,14 @@
 */
 //==============================================================================
 
+// We've been having some problems with continuous integration on Windows
+// making sure the stack is big enough to support test_nest_limits.  The
+// following lines are intended to make sure we get an adequate stack for
+// the test.
+#ifdef _MSC_VER
+#pragma comment(linker, "/STACK:4194304")
+#endif
+
```

Comment:
well, that's interesting, because the stack option is not mentioned in this list of settings that are available via this method:

https://msdn.microsoft.com/en-us/library/7f0aews7.aspx

Doesn't the stack size apply to the entire executable? If yes, then that would be rather pernicious to allow one compilation unit to change it. What would you do if different CUs set different values?

In any event, if the /STACK size does indeed apply to the executable as a whole, then I would strongly suggest it go here instead:

https://github.com/ripple/rippled/blob/eaff9a0e6aec0ad077f118501791c7684debcfd5/Builds/CMake/CMakeFuncs.cmake#L682

strictly speaking, it should also go in some similar place in the SConstruct I guess.

If I'm wrong and somehow stack size can apply to only one compilation unit, then I guess this pragma is appropriate.

Reply:
Yeah, I suspect the setting applies to the entire executable.  It also feels weird to me to set the stack size to support one specific unit test.  I'm not even confident this is a _real_ fix.  I just know it was failing Appveyor before and has Appveyor has passed unit tests twice with this change.

Reply:
Having slept on it over the weekend, I'm getting less comfortable with this proposed solution.  The reasons are two fold:
1. For coroutines we only require a 1 Meg (not 4 Meg) stack: https://github.com/ripple/rippled/blob/develop/src/ripple/core/Coro.ipp#L43.  If we're going to require a minimum stack size we should at least be consistent.
2. The reason for a recursion limit is so the user sees a reasonable error message rather than crashing the application.  Inflating the stack size only for the benefit of the test seems like the tail wagging the dog.  The test (and the allowed recursion depth) should suit itself to the environment we expect to be running in.

So, at this point, setting a minimum stack size seems reasonable since coroutines require one.  But we should be using the same minimum stack size everywhere (unless someone can provide a documentable reason for an exception).  Since coroutines are running with a 1 Meg stack, it seems like that should be the minimum for all stacks.

I also agree with @mellery451 that the right place to be enforcing a minimum stack size is in the various makefiles.  Then, if we have recursion depth crashes with a minimum stack size of 1 Meg we should either increase all minimum stack sizes or reduce the allowed recursion depth.  Just my feeling at the moment.

Thoughts?

Reply:
I don't believe this is a permanent solution, it was only meant for testing with Appveyor. A better temporary solution is lowering the nest limit as we discussed. I've written to the Appveyor support team and hopefully, they will provide some guidance on determining the difference we are seeing between their platform and the Windows desktops.

Reply:
AppVeyor replied and suggested the problem may be due to the Hyper-V VM we are using, which is configured for dynamic memory. They enabled GCE VMs which use standard memory and suggested we give that a whirl. @scottschurr You can try that by editing the YAML file and adding `appveyor_build_worker_cloud: gce` under the environment: section

Reply:
Thanks @miguelportilla.  I'll give that a whirl and report back.  At the same time, I'm also reducing the `json_reader` recursion depth to 25, since that approach was agreed up on earlier.  I figure anything we can do to stabilize Appveyor until we complete the transition to Jenkins will be good for now.